lib/pure/concurrency/threadpool.nim

```diff
@@ -224,6 +259,7 @@ proc nimFlowVarSignal(fv: FlowVarBase) {.compilerProc.} =
     signal(fv.ai.cv.c)
   if fv.usesSemaphore:
     signal(fv.cv)
+  fv.event.trigger()
 
 proc awaitAndThen*[T](fv: FlowVar[T]; action: proc (x: T) {.closure.}) =
```

Comment:
We need to deprecate these. If you have some time please create a seperate PR to do that :)